### PR TITLE
gate - error handling

### PR DESCRIPTION
**Pull request by pcriadoperez**
- fix docstring
- check required credentials for private endpoints
- Unify symbols when calling `fetchPositions`
- Throw better error message when calling `fetchPositions` with the wrong type

### CODE REVIEW

File: js/gate.js [hunk 1]
```diff
@@ -4102,7 +4102,11 @@ module.exports = class gate extends Exchange {
          * @returns {[object]} a list of [position structure]{@link https://docs.ccxt.com/en/latest/manual.html#position-structure}
          */
         await this.loadMarkets ();
+        symbols = this.marketSymbols (symbols);
         const [ type, query ] = this.handleMarketTypeAndParams ('fetchPositions', undefined, params);
+        if (type !== 'swap' && type !== 'future') {
```
Comment (carlosmiei) [inline js/gate.js:4107]: @pcriadoperez if the user provides one or more symbols that should be used to infer the type and currently we're ignoring it

**Comment by pcriadoperez** [comment]
I fixed the comments... also should this endpoint be included in `fetchPositions` for markets of type option?
https://www.gate.io/docs/developers/apiv4/en/#list-user-s-positions-of-specified-underlying

**Comment by carlosmiei** [comment]
> I fixed the comments... also should this endpoint be included in `fetchPositions` for markets of type option? https://www.gate.io/docs/developers/apiv4/en/#list-user-s-positions-of-specified-underlying

@pcriadoperez yes but on a different PR please (to avoid blocking this one)

**Comment by carlosmiei** [comment]
LGTM ✅ 

```
 p gate fetchPositions '["ADA/USDT:USDT"]' --swap
Python v3.10.6
CCXT v1.95.43
gate.fetchPositions(['ADA/USDT:USDT'])
[{'collateral': 1.8668895,
  'contractSize': 10.0,
  'contracts': 10.0,
  'datetime': None,
  'entryPrice': 0.36786,
  'id': None,
  'info': {'adl_ranking': '5',
           'close_order': None,
           'contract': 'ADA_USDT',
           'cross_leverage_limit': '5',
           'entry_price': '0.36786',
           'history_pnl': '-1.465078707',
           'history_point': '0',
           'last_close_pnl': '-0.500087244',
           'leverage': '20',
           'leverage_max': '20',
           'liq_price': '0.35843',
           'maintenance_rate': '0.025',
           'margin': '1.8668895',
           'mark_price': '0.36892',
           'mode': 'single',
           'pending_orders': '0',
           'realised_pnl': '-0.018393',
           'realised_point': '0',
           'risk_limit': '500000',
           'size': '10',
           'unrealised_pnl': '0.106',
           'user': '10406147',
           'value': '36.892'},
  'initialMargin': 1.872269,
  'initialMarginPercentage': 0.05075,
  'leverage': 20.0,
  'liquidationPrice': 0.35843,
  'maintenanceMargin': 0.9223,
  'maintenanceMarginPercentage': 0.025,
  'marginMode': 'isolated',
  'marginRatio': None,
  'markPrice': 0.36892,
  'notional': 36.892,
  'percentage': 5.661579612758636,
  'side': 'long',
  'symbol': 'ADA/USDT:USDT',
  'timestamp': None,
  'unrealizedPnl': 0.106}]
```

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**